javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/CompilationUnitContext.java

```diff
@@ -85,9 +77,10 @@ public CompilationUnitContext(CompilationUnit wrappedNode, TypeSolver typeSolver
         while (itName.contains(".")) {
             String typeName = getType(itName);
             String memberName = getMember(itName);
-            SymbolReference<ResolvedTypeDeclaration> type = this.solveType(typeName);
-            if (type.isSolved()) {
-                return new SymbolSolver(typeSolver).solveSymbolInType(type.getCorrespondingDeclaration(), memberName);
+            Optional<? extends ResolvedTypeDeclaration> type = this.solveType(typeName)
```

Comment:
note that this `type` should be named `declaration` now, given that it's the (optional) declaration being returned rather than the type :+1: 

there are multiple places this applies